### PR TITLE
Component is now just a type synonym

### PR DESCRIPTION
**Pull request by paf31**
@garyb @jdegoes @puffnfresh @cryogenian Could you please review?

This doesn't _remove_ `Component`, as I originally suggested, but turns it into a type synonym (minus the existential `i` part), and keeps some of the functions around, since they are still useful.

Things worth noticing:

- The examples didn't change much, mostly just removing calls to `component`
- The type of `install` has changed. Making the installed component dependent on the placeholder value means we have to have `HTML` on the RHS, not a whole `Component`, but I think this is okay, since the function will close over the state of the outside component in practice. 

I think it's worth thinking about putting together a library of isomorphisms (`Either`, `Tuple` and `Maybe` should do it) to deal with the few more `Either`s we'll see as a result of ditching the existential.

### CODE REVIEW

**Comment by paf31** [comment]
Resolves #84, #99.

**Comment by cryogenian** [comment]
I like it. Code is simpler, probably more performant and this simplification doesn't remove any important functionality. :+1:

**Comment by paf31** [comment]
I started a bit of work on an isomorphisms library, and I'll work on it some more this weekend:

https://github.com/paf31/purescript-isomorphisms/

**Comment by puffnfresh** [comment]
The new `install` definitely seems a lot more useful. Hopefully it will fix my blocker.

File: src/Halogen/Component.purs [hunk 1]
```diff
@@ -183,25 +124,7 @@ combine :: forall p q r m req1 req2 res1 res2.
              Component p m req1 res1 -> 
              Component q m req2 res2 -> 
              Component r m (Either req1 req2) (Either res1 res2)
-combine f = runComponent \sf1 -> runComponent \sf2 -> component' (mergeWith' f1 f2 sf1 sf2)
+combine f = mergeWith f1
```
Comment (puffnfresh) [inline src/Halogen/Component.purs]: This is great. The previous stuff was really bad :fireworks:

**Comment by puffnfresh** [comment]
This cleans things up dramatically. I'm not sure if my blocker will be fixed, but :shipit:

**Comment by jdegoes** [comment]
Big improvement. :+1:

**Pull request merged**